### ISSUE TITLE
Potentially misleading Dropout used in MNIST example

### ISSUE THREAD

**Issue by alik604** [pytorch/examples#805]
The [MNIST example](https://github.com/pytorch/examples/blob/master/mnist/main.py) has the following
``` Python 
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = nn.Conv2d(1, 32, 3, 1)
        self.conv2 = nn.Conv2d(32, 64, 3, 1)
        self.dropout1 = nn.Dropout2d(0.25)
        self.dropout2 = nn.Dropout2d(0.5)          # <---  See this 
        self.fc1 = nn.Linear(9216, 128)
        self.fc2 = nn.Linear(128, 10)

    def forward(self, x):
        x = self.conv1(x)
        x = F.relu(x)
        x = self.conv2(x)
        x = F.relu(x)
        x = F.max_pool2d(x, 2)
        x = self.dropout1(x)
        x = torch.flatten(x, 1)                    # <--- See this 
        x = self.fc1(x)
        x = F.relu(x)
        x = self.dropout2(x)                       # <--- See this, I think it should be a `nn.Dropout`
        x = self.fc2(x)
        output = F.log_softmax(x, dim=1)
        return output
```

The docs do say "**Usually** the input comes from nn.Conv2d modules.", however I'm not sure if the example is following best practices and is beginner friendly 

... Just thought I should bring this up. I'm a TF user switching to PyTorch. This issue is likely more a matter of form and understandably, than it is of technical correctness (I results are fine...)

**Comment by msaroufim** [comment]
Seems OK to me, feel free to make a PR if you'd like to see this change in
